cmd/generate/config/rules/kubernetes.go

```diff
@@ -0,0 +1,68 @@
+package rules
+
+import (
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+// The kubernetes rules are split into two functions to make the complex proximity matching of the data-key and the kind-identifier more readable and testable
+
+// KubernetesSecretWithDataBefore validates if we detected a kubernetes secret which contains data, before the resource identifier!
+func KubernetesSecretWithDataBefore() *config.Rule {
+	// define rule
+	r := config.Rule{
+		RuleID:      "kubernetes-secret-with-data-before",
+		Description: "Possible Kubernetes Secret detected, posing a risk of leaking credentials/tokens from your deployments",
+		// We try to match secrets by looking if we have the keyword
+		Regex: generateUniqueTokenRegex(`(?i)(?:\b(data:))(\W+(?:\w+\W+){0,200}?)\bkind:.*Secret\b`, true),
```

Comment:
Data doesn't need to be a capturing group.
```suggestion
		Regex: generateUniqueTokenRegex(`(?i)(?:\b(?:data:))(\W+(?:\w+\W+){0,200}?)\bkind:.*Secret\b`, true),
```
-----
Also, I'm apprehensive about using `.*` between `kind:` and `secret`. As far as I'm aware, the only valid characters would be whitespace and an optional quote (`"'`).

I think it would be best to either limit the characters to known valid ones, or put a reasonable limit to reduce any false positives (e.g., `.{0,10}`).

Reply:
> Also, I'm apprehensive about using .* between kind: and secret. As far as I'm aware, the only valid characters would be whitespace and an optional quote ("').
> I think it would be best to either limit the characters to known valid ones, or put a reasonable limit to reduce any false positives (e.g., .{0,10}).

@localleon 
I agree with this. A `.{0,10}` should do the trick like @rgmz suggested.